appsec/appsec.go

```diff
@@ -30,3 +33,92 @@ func MonitorParsedHTTPBody(ctx context.Context, body interface{}) {
 	}
 	// bonus: use sync.Once to log a debug message once if AppSec is disabled
 }
+
+// TrackUserLoginSuccessEvent sets a successful user login event, with the given
+// user id and optional metadata, as service entry span tags. It also calls
+// tracer.SetUser() to set the currently identified user, along the given
+// tracer.UserMonitoringOption options.
+// The service entry span is obtained through the given Go context which should
+// contain the currently running span. This function does nothing when no span
+// is found in the given Go context, along with an error message.
+// Such events trigger the backend-side events monitoring, such as the Account
+// Take-Over (ATO) monitoring, ultimately blocking the IP address and/or user id
+// associated to them.
+func TrackUserLoginSuccessEvent(ctx context.Context, uid string, md map[string]string, opts ...tracer.UserMonitoringOption) {
+	span := getRootSpan(ctx)
+	if span == nil {
+		return
+	}
+
+	const tagPrefix = "appsec.events.users.login.success."
+	span.SetTag(tagPrefix+"track", true)
+	for k, v := range md {
+		span.SetTag(tagPrefix+k, v)
+	}
+	span.SetTag(ext.SamplingPriority, ext.PriorityUserKeep)
+	tracer.SetUser(span, uid, opts...)
+}
+
+// TrackUserLoginFailureEvent sets a failed user login event, with the given
+// user id and the optional metadata, as service entry span tags. The exists
+// argument allows to distinguish whether the given user id actually exists or
+// not.
+// The service entry span is obtained through the given Go context which should
+// contain the currently running span. This function does nothing when no span
+// is found in the given Go context, along with an error message.
```

Comment:
Same suggestion as above:
```suggestion
// is found in the given Go context and logs an error message instead.
```